python/cudf/dataframe/string.py

```diff
@@ -108,9 +108,43 @@ def cat(self, others=None, sep=None, na_rep=None):
             (same type as caller) of str dtype is returned.
         """
         from cudf.dataframe import Series, Index
+        def is_list_like(others):
+            if isinstance(others, (Series, list, tuple, Index)) and not isinstance(others, (str, bytes)):
+                return True
+            else:
+                return False
+
         if isinstance(others, (Series, Index)):
+            # If others is just another Series/Index, great go ahead with concatenation
             assert others.dtype == np.dtype('object')
             others = others.data
+        elif is_list_like(others):
+            # If others is a list-like object (in our case lists & tuples)just another Series/Index, great go ahead with concatenation
+            if all(is_list_like(x) for x in others):
```

Comment:
Instead of checking if everything is a list-like, I think we should just check if the first element is a string, if it is we can convert to an nvstrings object to pass to the underlying function, if it’s not then we can assume we’re dealing with a list-like of list-likes and convert each object to an nvstrings object and run the underlying function.

Reply:
As this is, it will be super expensive or break if we give a large array-like of strings, especially if it’s on the GPU.

Reply:
But `others` can be a strings list, strings tuple or even a list which is mix of Series & strings list(like below) in any order:

`[['f','a','b','f','a'], pd.Series(['f', 'g', 'h', 'i', 'j']),['f','a','b','f','a']]`


So, if we proceed without checking all of `others` types and passing to nvstrings it would either crash or lead the kernel to hand. Please correct me if I understood something wrong.